src/lavinmq/clustering/client.cr

```diff
@@ -19,14 +71,28 @@ module LavinMQ
       def initialize(@config : Config, @id : Int32, @password : String, proxy = true)
         System.maximize_fd_limit
         @data_dir = config.data_dir
-        @files = Hash(String, File).new do |h, k|
-          path = File.join(@data_dir, k)
-          Dir.mkdir_p File.dirname(path)
-          h[k] = File.open(path, "a").tap &.sync = true
```

Comment:
If you just changed this line instead, to sync false. What kind of performance do you then get?

Reply:
Seems to be on par with m files.